tictactoe.js

```diff
@@ -0,0 +1,144 @@
+const readline = require("readline");
+
+const readlineInterface = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Ask question
+const ask = questionText => {
+  return new Promise((resolve, reject) => {
+    readlineInterface.question(questionText, resolve);
+  });
+};
+
+let values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+let currentPlayer = "X";
+let winner = "";
+
+const grid = gridValues => {
+  return (
+    "\n" +
+    " " +
+    gridValues[1] +
+    " | " +
+    gridValues[2] +
+    " | " +
+    gridValues[3] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[4] +
+    " | " +
+    gridValues[5] +
+    " | " +
+    gridValues[6] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[7] +
+    " | " +
+    gridValues[8] +
+    " | " +
+    gridValues[9] +
+    "\n\n" +
+    "Player " +
+    currentPlayer +
+    "'s turn\n" +
+    "Move to?\n"
+  );
+};
+
+const checkWinner = player => {
+  if (
+    (values[1] === player && values[2] === player && values[3] === player) ||
```

Comment:
think about how to write this by looping over a data structure instead of repeating `(values[ ] === player && values[ ] === player && values[ ] === player)` a dozen times